packages/material-ui/src/Stack/Stack.js

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {
+  createUnaryUnit,
+  handleBreakpoints,
+  unstable_extendSxProp as extendSxProp,
+} from '@material-ui/system';
+import { getValue } from '@material-ui/system/spacing';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+
+const getSideFromDirection = (direction) => (direction === 'row' ? 'Left' : 'Top');
+
+const StackRoot = experimentalStyled(
+  'div',
+  {},
+  { name: 'Stack' },
+)(({ styleProps, theme }) => {
+  let styles = {
+    display: 'flex',
+    flexDirection: styleProps.direction,
+  };
+
+  if (styleProps.spacing) {
+    const transformer = createUnaryUnit(theme, 'spacing', 8, 'spacing');
+    const styleFromPropValue = (propValue) => ({
+      '& > * + *': {
+        [`margin${getSideFromDirection(styleProps.direction)}`]: getValue(transformer, propValue),
+      },
+    });
+
+    styles = {
+      ...styles,
+      ...handleBreakpoints({ theme }, styleProps.spacing, styleFromPropValue),
+    };
+  }
+  return styles;
+});
+
+const Stack = React.forwardRef((inProps, ref) => {
```

Comment:
To get the correct messages in React dev tools

```suggestion
const Stack = React.forwardRef(function Stack(inProps, ref) => {
```